core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -60,7 +61,7 @@ class LogManager(logDirs: Seq[File],
                  val retentionCheckMs: Long,
                  val maxPidExpirationMs: Int,
                  scheduler: Scheduler,
-                 val brokerState: BrokerState,
+                 val brokerState: AtomicReference[BrokerState],
```

Comment:
Is there a reason why you're using an AtomicReference instead of a volatile var?

Reply:
The log manager needs to alter the broker state using the `AtomicReference`.  Since the broker state is owned by `KafkaBroker`, this would not be possible by just using a volatile variable.

(This is an ugly pattern, I think, and a later patch set removes it.  But in the interest of keeping this PR small, we preserve the original behavior where LogManager changes the broker state directly for now)